### PR TITLE
MAP-1448 LocationResource file split accordingly with Test split MAP-…

### PR DESCRIPTION
**Pull request by marcusaleman**
1 LocationInt.kt
Get Location info for {Id} and activate and deactivate.     Activate and de-Activate residential Location will be split later accordingly with the tests.

    @RequestMapping(“/locations" 
    @RequestMapping(“/locations/{id}”  
    @PutMapping("/{id}/deactivate/temporary")
    @PutMapping("/{id}/reactivate")
    @PutMapping("/{id}/update/temporary-deactivation")

    2 LocationKeyt.kt 
    Endpoints Manipulates Keys

   @GetMapping(“/key/{key}")
   @PostMapping("/keys", produces = [MediaType.APPLICATION_JSON_VALUE])
    
    3 LocationPrisonId.kt 
    Get information related with PrisonID (establishment)

    @GetMapping("/prison/{prisonId}")
    @GetMapping("prison/{prisonId}/groups")
    @GetMapping("prison/{prisonId}/group/{group}/location-prefix")
 
    @GetMapping("/prison/{prisonId}/inactive-cells")
    @GetMapping("/prison/{prisonId}/usage-type/{usageType}")
    @GetMapping("/prison/{prisonId}/location-type/{locationTYpe}”)
	
    4 LocationTransform.kt 
    Transform a LocationID (any kind of transformation) 
    
   @PutMapping("/{id}/used-for-type", produces = [MediaType.APPLICATION_JSON_VALUE]
   @PutMapping("/{id}/specialist-cell-types", produces = [MediaType.APPLICATION_JSON_VALUE])
   @PutMapping("/{id}/capacity")
    
    5 LocationResidential.kt DONE
    Residential and Non Residential manipulation. 
    Convert Cell Non Residential to Residential, etc.
    Create wing res or non res.

   @GetMapping("/residential-summary/{prisonId}")
   @PostMapping("/create-wing", produces = [MediaType.APPLICATION_JSON_VALUE]) 
   @PatchMapping("/residential/{id}")
   @PostMapping("/residential", produces = [MediaType.APPLICATION_JSON_VALUE])
   @PatchMapping(“/residential/{id}")
   @PostMapping("/non-residential", produces = [MediaType.APPLICATION_JSON_VALUE])
   @PutMapping(“/{id}/convert-cell-to-non-res-cell”)
   @PutMapping("/{id}/convert-to-cell")

### CODE REVIEW

**Review by Mjwillis — CHANGES_REQUESTED** [review]
All fine except should call them Resource at the end of the class name

**Comment by marcusaleman** [comment]
@Mjwillis 
I have reopen the PR. May you check if it is ok, please?

**Comment by marcusaleman** [comment]
hi, @Mjwillis  i need to go through with this PR with you to clear that part as i understand that it is a massive PR

File: src/test/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/jpa/repository/LocationRepositoryTest.kt [hunk 1]
```diff
@@ -137,7 +137,7 @@ class LocationRepositoryTest : TestBase() {
     assertThat(cell3Converted.isCertified()).isFalse()
 
     cell3Converted.convertToCell(
-      accommodationType = AllowedAccommodationTypeForConversion.NORMAL_ACCOMMODATION,
+      accommodationType = LocationResidentialResource.AllowedAccommodationTypeForConversion.NORMAL_ACCOMMODATION,
```
[the remark this answers is not in the document]
Comment (marcusaleman) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/jpa/repository/LocationRepositoryTest.kt]: @Mjwillis 
Enum import fixed AllowedAccommodationTypeForConversion

**Review by Mjwillis — APPROVED** [review]
I'll merge

**Pull request merged**